packages/ui/src/components/utils/popover/index.tsx

```diff
@@ -52,25 +52,34 @@ export const Popover = forwardRef<HTMLDivElement, PopoverProps>(
     ): ReactElement {
         const [popper, setPopper] = useState<HTMLDivElement | null>(null);
 
-        useEffect(() => {
-            if (!popper || !ref) return;
-            if (typeof ref === "function") ref(popper);
-            else ref.current = popper;
-        }, [popper, ref]);
+        const { styles, attributes, update, forceUpdate } = usePopper(
+            anchor,
+            popper,
+            {
+                placement,
+                modifiers: [
+                    {
+                        name: "offset",
+                        options: { offset },
+                    },
+                ],
+            }
+        );
+        console.log(anchor, styles, attributes, update, forceUpdate);
 
-        const { styles, attributes } = usePopper(anchor, popper, {
-            placement,
-            modifiers: [
-                {
-                    name: "offset",
-                    options: { offset },
-                },
-            ],
-        });
+        useLayoutEffect(() => {
+            if (update) update();
+        }, [update]);
 
         return (
             <div
-                ref={setPopper}
+                ref={(element) => {
+                    if (ref) {
+                        if (typeof ref === "function") ref(element);
```

Comment:
Why do we need to check if the `ref` is a `function` here?

Reply:
TS complains otherwise, since you can also pass functions to `ref` (that's actually what we're doing here after all with the arrow function).